integration-tests/itest/deploy_suite_test.go

```diff
@@ -0,0 +1,13 @@
+package main
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestDeploy(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Deploy Suite")
+}
```

Comment:
For me, not being a Gingko expert, it is not clear how this line somehow runs the tests in `deploy_test.go`. How are those two linked? By name? By convention?

Reply:
It might be useful for us non-Golang experts to add a comment explaining how to work with this in the future.

Reply:
I hope a Go expert can explain. This is auto-generated by the framework. 
(Intuitively, I think that this line hooks into the built-in test support by registering one test called "TestDeploy" and uses that to bootstrap its own runtime.)

Reply:
@agemooij the [ginkgo home](https://onsi.github.io/ginkgo/) page has a breakdown of this structure under the 'getting started' head. 
>Let’s break this down:

Go allows us to specify the books_test package alongside the books package. Using books_test instead of books allows us to respect the encapsulation of the books package: your tests will need to import books and access it from the outside, like any other package. This is preferred to reaching into the package and testing its internals and leads to more behavioral tests. You can, of course, opt out of this – just change package books_test to package books
We import the ginkgo and gomega packages into the test’s top-level namespace by performing a dot-import. If you’d rather not do this, check out the Avoiding Dot Imports section below.
TestBooks is a testing test. The Go test runner will run this function when you run go test or ginkgo.
RegisterFailHandler(Fail): A Ginkgo test signals failure by calling Ginkgo’s Fail(description string) function. We pass this function to Gomega using RegisterFailHandler. This is the sole connection point between Ginkgo and Gomega.
RunSpecs(t *testing.T, suiteDescription string) tells Ginkgo to start the test suite. Ginkgo will automatically fail the testing.T if any of your specs fail.

Reply:
I don't see a need to replicate any of that on our code.